doc/Programming-guide/Key-platform-features/Common-Type-Library/index.md

```diff
@@ -5,17 +5,51 @@ permalink: /:path/
 nav: /:path/Programming-guide/Key-platform-features/Common-Type-Library/
 sort_idx: 20
 ---
+{% assign root_url = page.url | split: '/'%}
+{% capture root_url  %} /{{root_url[1]}}/{{root_url[2]}}/{% endcapture %}
 
-Common Type Library
+## Common Type Library ##
 
-* [CT scopes](#ct-scopes)
+* [CT types definitions and FQNs](#ct-types-definitions-and-fqns)
 * [CT schema versioning and dependencies](#ct-schema-versioning-and-dependencies)
-* [CT schema export support](#ct-schema-export-support)
+* [CT scopes](#ct-scopes)
+* [CT management](#ct-management)
+* [CT promotion](#ct-promotion)
+* [CT schema import and export](#ct-schema-import-and-export)
 
 
 The Common Type Library (**CTL**) is a repository of reusable data type schemas that could be referenced and reused as the user creates specific schema definitions for Kaa modules.
 This capability unifies management of all schemas in a Kaa server instance.
 
+## CT types definitions and FQNs ##
+
+CT schema instance is identified by fully qualified name (FQN) and version. FQN is combination of namespace and name attributes defined in root avro record of CT schema according to [Names in Avro  Schema Declaration](http://avro.apache.org/docs/current/spec.html#names). After creation CT becomes unmodifiable, only new version of CT with same FQN can be created afterwards.
```

Comment:
*CT schema instance* is not self-explanatory. How does it differ from *CT* or *CT schema*? Please choose the terminology, explain it and stick to it everywhere.